### PR TITLE
Tests for validatePayload, generateExtensionsEntry, and ensureExtensionsDir

### PR DESCRIPTION
**Pull request by jamescammarano**
- validatePayload tests passing 100% coverage 
- generateExtensionsEntry tests passing 100% coverage 
- ensureExtensionsDir tests passing missing line 12 in stats but appears to be used in the test "throws an error when a folder cant be opened"
- added `tmp` package to dev dependencies

### CODE REVIEW

File: packages/shared/src/utils/node/ensure-extension-dirs.test.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import { EXTENSION_TYPES } from '../../constants/extensions';
+import { ExtensionType } from '../../types';
+
+describe('ensureExtensionDirs', () => {
+	const types = EXTENSION_TYPES as readonly ExtensionType[];
+	it('returns undefined if the folders exist', async () => {
+		expect(await ensureExtensionDirs('./extension', types)).toBe(undefined);
+	});
+
+	it('throws an error when a folder cant be opened', () => {
```
Comment (rijkvanzanten) [inline packages/shared/src/utils/node/ensure-extension-dirs.test.ts:11]: We should add some before/after functions here to setup/teardown the folder structure to test, to ensure other tests/repeated invocations don't cause state-based inconsistencies in the test results

**Review by rijkvanzanten — CHANGES_REQUESTED** [review]

File: packages/shared/src/utils/node/ensure-extension-dirs.test.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+describe('ensureExtensionDirs', () => {
+	const types = EXTENSION_TYPES as readonly ExtensionType[];
+	it('returns undefined if the folders exist', async () => {
+		expect(await ensureExtensionDirs(rootDir.name, types)).toBe(undefined);
+	});
+
+	it('throws an error when a folder cant be opened', () => {
+		expect(async () => {
+			await ensureExtensionDirs('/.', types);
+		}).rejects.toThrow(`Extension folder "/interfaces" couldn't be opened`);
```
Comment (rijkvanzanten) [inline packages/shared/src/utils/node/ensure-extension-dirs.test.ts:25]: This `/interfaces` name is based on the imported const, but isn't really specific to this particular test/function. we should test with a custom name so we can test the function, rather then test the function combined with the external `types` array (the function should work regardless of input array)

```suggestion
			await ensureExtensionDirs('/.', ['test']);
		}).rejects.toThrow(`Extension folder "/test" couldn't be opened`);
```

**Review by rijkvanzanten — CHANGES_REQUESTED** [review]

**Comment by rijkvanzanten** [comment]
Gotta keep things as decoupled as possible for these unit tests 🙂

File: packages/shared/package.json [hunk 3]
```diff
@@ -59,6 +59,7 @@
 		"knex-schema-inspector": "*",
 		"lodash": "4.17.21",
 		"pino": "*",
+		"tmp": "^0.2.1",
```
Comment (nickrum) [inline packages/shared/package.json:62]: This should probably be a dev dependency.

**Review by nickrum — CHANGES_REQUESTED** [review]

**Review by rijkvanzanten — CHANGES_REQUESTED** [review]

**Review by nickrum — APPROVED** [review]

**Review by rijkvanzanten — APPROVED** [review]

**Pull request merged**